receiver/zipkinreceiver/trace_receiver.go

```diff
@@ -132,30 +133,20 @@ func (zr *ZipkinReceiver) v2ToTraceSpans(blob []byte, hdr http.Header) (reqs pda
 	//      https://github.com/openzipkin/zipkin-go/blob/3793c981d4f621c0e3eb1457acffa2c1cc591384/proto/v2/zipkin.proto#L154
 	debugWasSet := hdr.Get("X-B3-Flags") == "1"
 
-	var zipkinSpans []*zipkinmodel.SpanModel
+	// By default, we'll assume using JSON
+	unmarshaler := zr.jsonUnmarshaler
 
 	// Zipkin can send protobuf via http
-	switch hdr.Get("Content-Type") {
-	// TODO: (@odeke-em) record the unique types of Content-Type uploads
-	case "application/x-protobuf":
-		zipkinSpans, err = zipkin_proto3.ParseSpans(blob, debugWasSet)
-
-	default: // By default, we'll assume using JSON
-		zipkinSpans, err = zr.deserializeFromJSON(blob)
-	}
-
-	if err != nil {
-		return pdata.Traces{}, err
+	if hdr.Get("Content-Type") == "application/x-protobuf" {
+		// TODO: (@odeke-em) record the unique types of Content-Type uploads
+		if debugWasSet {
```

Comment:
Can you please investigate the reason why this was added and maybe consider to remove it if no longer useful.

Reply:
Zipkin uses debug=true to indicate that sampling decisions is overriden but I'm not sure why it is recorded as a part of the span. I assume, it allows developers to query traces initiated for debugging purposes. It's a part of their span model.

Reply:
@bogdandrutu does this resolve it?

Reply:
That is for in process propagation as far as I know, not for collection. This is where it is unclear to me